backend/src/tools/openapi_spec.py

```
@@ -0,0 +1,122 @@
+        
+    Raises:
+        Exception: If the request fails or returns a non-200 status code
+    """
+    try:
+        response = httpx.get(url)
+        if response.status_code == 200:
+            return yaml.dump(response.json(), sort_keys=False)
+        else:
+            raise Exception(f"Failed to get OpenAPI spec from {url}")
+    except Exception as e:
+        raise Exception(f"Failed to get OpenAPI spec from {url}: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)